src/Core/Renderer.php

```diff
@@ -66,28 +66,27 @@ class Renderer
 	];
 
 	/**
-	 * This is our template processor
+	 * Returns the rendered template output from the template string and variables
 	 *
-	 * @param string|FriendicaSmarty $s    The string requiring macro substitution or an instance of FriendicaSmarty
-	 * @param array                  $vars Key value pairs (search => replace)
-	 *
-	 * @return string substituted string
-	 * @throws Exception
+	 * @param string $template
+	 * @param array  $vars
+	 * @return string
+	 * @throws InternalServerErrorException
 	 */
-	public static function replaceMacros($s, array $vars = [])
+	public static function replaceMacros(string $template, array $vars)
 	{
 		$stamp1 = microtime(true);
 
 		// pass $baseurl to all templates if it isn't set
-		$vars = array_merge(['$baseurl' => DI::baseUrl()->get()], $vars);
+		$vars = array_merge(['$baseurl' => DI::baseUrl()->get(), '$APP' => DI::app()], $vars);
 
 		$t = self::getTemplateEngine();
 
 		try {
-			$output = $t->replaceMacros($s, $vars);
+			$output = $t->replaceMacros($template, $vars);
 		} catch (Exception $e) {
-			echo "<pre><b>" . __FUNCTION__ . "</b>: " . $e->getMessage() . "</pre>";
-			exit();
+			DI::logger()->critical($e->getMessage(), ['template' => $template, 'vars' => $vars]);
```

Comment:
Why "critical" and not "error"? I would define "critical" as some "core meltdown is expected to happen 10 seconds" event.

Reply:
According to the logger functions in the PSR `LoggerInterface`:
```php
    /**
     * Critical conditions.
     *
     * Example: Application component unavailable, unexpected exception.
     *
     * @param string $message
     * @param array  $context
     *
     * @return void
     */
    public function critical($message, array $context = array());

    /**
     * Runtime errors that do not require immediate action but should typically
     * be logged and monitored.
     *
     * @param string $message
     * @param array  $context
     *
     * @return void
     */
    public function error($message, array $context = array());
```
The templating engine failing is an "application component unavailable" and certainly not an "error that does not require immediate action".

What you are describing is either "alert" or "emergency", the two highest log levels even above "critical".

Reply:
Okay.